File: src/facade/examples/imagesearch/ImageSearchTest.js
'use strict';

const ImageSearchFacade = require('./ImageSearchFacade');
const RawImage = require('../../../adapter/examples/metadata/RawImage');
const SearchIntent = require('../../../strategy/examples/searchintent/SearchIntent');

const facade = new ImageSearchFacade();
const img = new RawImage([1, 2, 3, 4], 'JPEG');

console.log(facade.uploadAndSearch(img, SearchIntent.ADOPTION).toString());
console.log(facade.uploadAndSearch(img, SearchIntent.SALE).toString());
console.log(facade.uploadAndSearch(img, SearchIntent.VERIFY_LOSS).toString());

File: src/facade/examples/imagesearch/ImageSearchTest.java
package facade.examples.imagesearch;

import adapter.examples.metadata.RawImage;
import strategy.examples.searchintent.SearchIntent;

public class ImageSearchTest {
    public static void main(String[] args) {
        ImageSearchFacade facade = new ImageSearchFacade();
        RawImage img = new RawImage(new byte[]{1, 2, 3, 4}, "JPEG");

        System.out.println(facade.uploadAndSearch(img, SearchIntent.ADOPTION));
        System.out.println(facade.uploadAndSearch(img, SearchIntent.SALE));
        System.out.println(facade.uploadAndSearch(img, SearchIntent.VERIFY_LOSS));
    }
}
